core/src/state.rs

```diff
@@ -729,12 +836,10 @@ impl WorldTransaction<'_, '_> {
     /// # Errors
     /// If domain, account or asset not found
     pub fn asset_mut(&mut self, id: &AssetId) -> Result<&mut Asset, FindError> {
-        self.account_mut(&id.account).and_then(move |account| {
-            account
-                .assets
-                .get_mut(&id.definition)
-                .ok_or_else(|| FindError::Asset(id.clone()))
-        })
+        let _ = self.account_mut(&id.account)?;
```

Comment:
```suggestion
        let _ = self.account(&id.account)?;
```

If there is no need for changing account let's not use mutable methods.

Currently because there is no way to find out that you want change mutable reference`Storage` would create clone of previous account value, so that it can revert changes in case of transaction failure.